### PR TITLE
feat: updated to include async loading

### PR DESCRIPTION
**Pull request by SGFGOV**
Continuation of feat: added async loading support to medusa #139
fixed merging issues

### CODE REVIEW

**Review by adrien2p — CHANGES_REQUESTED** [review]
LGTM, some linting issues with the fixtures files and one minor comment otherwise it will be good to go 🎉 thanks for you great effort

File: integration-tests/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,7242 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
[the remark this answers is not in the document]
Comment (SGFGOV) [reply integration-tests/yarn.lock]: done

**Comment by SGFGOV** [comment]
@adrien2p , thanks a ton for merging this and being extremely patient with me, please ignore the revert request. accidental click

**Comment by adrien2p** [comment]
@SGFGOV it is a pleasure. I am planning to release in the near future 💪 thanks for your work 😍

**Pull request merged**